### ISSUE TITLE
Unsetting extensions from Yaml does not seem to work

### ISSUE THREAD

**Issue by micschk** [silverstripe/silverstripe-framework#10511]
The below Yaml config produces an error: "Uncaught InvalidArgumentException: SilverStripe\Assets\File references nonexistent in 'extensions'".

```yaml
SilverStripe\Assets\File:
  extensions:
    comments: SilverStripe\Comments\Extensions\CommentsExtension
```
This would allow you to disable that extension with only this code

```yaml
SilverStripe\Assets\File:
  extensions:
    comments: null
```
_The above code example was originally posted by @tractorcow in https://github.com/silverstripe/silverstripe-cms/issues/2146#issuecomment-381803249_

## PRs
- https://github.com/silverstripe/silverstripe-framework/pull/10516

**Comment by GuySartorelli** [comment]
Fixed with #10516

**Comment by micschk** [comment]
Great, thx!
